### PR TITLE
Raphael Fritz Woche 6 (1778099 1623571)

### PR DESCRIPTION
**Pull request by FritzRa**

### CODE REVIEW

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
+#Usage of filepaths as keys because of directories with identical names but different sizes and filepaths.
+
+def get_terminal_session(filepath):     
+    #gives out the terminal_record.txt file as a list
+    with open(filepath, 'r') as file:
+        sessionlist = []
+        for line in file:
+            sessionlist.append(line.replace("\n",""))
+        return sessionlist
```
Comment (ClasherKasten) [inline Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:10]: This is essentially a list comprehension
```python
return [line.replace('\n', '') for line in file]
```

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+from pathlib import Path
```
Comment (ClasherKasten) [inline Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:1]: Why use `pathlib` if all you need it for is to pass it to `get_terminal_session`? Seems a bit bloated,

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    #gives out the terminal_record.txt file as a list
+    with open(filepath, 'r') as file:
+        sessionlist = []
+        for line in file:
+            sessionlist.append(line.replace("\n",""))
+        return sessionlist
+
+def get_filepath_deeper(sessionlist,index,filepath):    
+    #returns filepath to be used as a key of the subdirectory in dirdict
+    filepath = filepath + (sessionlist[index-1].replace("$ cd ","") + ("/"))
```
Comment (ClasherKasten) [inline Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:14]: Not OS independent. Either make use of the imported `pathlib` or even better: use `os.path.join`.

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    return filepath
+
+def get_filepath_parent(filepath):  
+    #returns filepath of the parent directory (in "$ cd .. " command)
+    filepath = "/".join(filepath.split('/')[0:-2]) + "/"
```
Comment (ClasherKasten) [inline Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:19]: Not OS independent. Either make use of the imported `pathlib` or even better: use `os.path.join`.

**Review by ClasherKasten — CHANGES_REQUESTED** [review]
Overall okay-ish, though some type annotations would be wonderful.

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py (hunk 3 above)
Comment (FritzRa) [reply Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:14]: the filepath in this function refers to the (fictional) filepath that is created when going through the session list, it therefore should have nothing to do with the OS.

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py (hunk 4 above)
Comment (FritzRa) [reply Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:19]: the filepath in this function refers to the (fictional) filepath that is created when going through the session list, it therefore should have nothing to do with the OS.

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py (hunk 3 above)
Comment (ClasherKasten) [reply Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:14]: You're right, thought i had thrown away this idea/comment...

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py (hunk 4 above)
Comment (FritzRa) [reply Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:15]: I am also wondering:
https://github.com/dhbw-ma-ppp/ppp-2023/pull/142#issuecomment-1812010539

File: Exercises/Raphael Fritz/Sechste Woche Aufgaben.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+
+def task_2(dirsizedict):
+    #returns smallest file size that if deleted would allow for 30000000 of free space
+    deletionlist = []
+    unused = 70000000 - dirsizedict["/"]
+    required = 30000000
+    for value in dirsizedict.values():
+        if unused + value >= required:
+            deletionlist.append(value)
+    return min(deletionlist)
```
Comment (FelixRegnier) [inline Exercises/Raphael Fritz/Sechste Woche Aufgaben.py:75]: Ich wuerde es bevorzugen, wenn die Grenzen fuer die Aufgaben als Argument uebergeben werden, ist in diesem Fall jedoch nicht falsch, da die Aufgabe es so vorgegeben hat

**Review by FelixRegnier — APPROVED** [review]
Insgesamt gute Loesung fuer die Aufgabe. Die Task-Funktionen wuerde ich allerdings nicht so hart coden.

**Pull request merged**